tools/azure-sdk-tools/devtools_testutils/azure_recorded_testcase.py

```diff
@@ -132,26 +132,31 @@ def get_credential(self, client_class, **kwargs):
                     from azure.identity.aio import AzureDeveloperCliCredential
                 return AzureDeveloperCliCredential()
 
-            # Service principal authentication
-            if tenant_id and client_id and secret:
-                # Check for track 2 client
-                if _is_autorest_v3(client_class):
-                    _LOGGER.info(
-                        "Service principal client ID, secret, and tenant ID detected. Using ClientSecretCredential.\n"
-                        "For user-based auth, set AZURE_TEST_USE_PWSH_AUTH or AZURE_TEST_USE_CLI_AUTH to 'true'."
-                    )
-                    from azure.identity import ClientSecretCredential
-
-                    if is_async:
-                        from azure.identity.aio import ClientSecretCredential
-                    return ClientSecretCredential(tenant_id=tenant_id, client_id=client_id, client_secret=secret)
-                else:
-                    # Create msrestazure class
-                    from msrestazure.azure_active_directory import (
-                        ServicePrincipalCredentials,
-                    )
-
-                    return ServicePrincipalCredentials(tenant=tenant_id, client_id=client_id, secret=secret)
+            from azure.identity import DefaultAzureCredential
+            if is_async:
+                from azure.identity.aio import DefaultAzureCredential
+            return DefaultAzureCredential()
+
+            # Stop using Service principal authentication. Use DefaultAzureCredential instead.
+            # if tenant_id and client_id and secret:
+            #     # Check for track 2 client
+            #     if _is_autorest_v3(client_class):
+            #         _LOGGER.info(
+            #             "Service principal client ID, secret, and tenant ID detected. Using ClientSecretCredential.\n"
+            #             "For user-based auth, set AZURE_TEST_USE_PWSH_AUTH or AZURE_TEST_USE_CLI_AUTH to 'true'."
+            #         )
+            #         from azure.identity import ClientSecretCredential
+            #
+            #         if is_async:
+            #             from azure.identity.aio import ClientSecretCredential
+            #         return ClientSecretCredential(tenant_id=tenant_id, client_id=client_id, client_secret=secret)
+            #     else:
+            #         # Create msrestazure class
+            #         from msrestazure.azure_active_directory import (
```

Comment:
Do you know if there are any existing clients that would still need `msrestazure` credential support? I would think that by now there may not be, but if there are any we should continue to support them

Reply:
I just double-checked and it looks like `azure-batch` is still on track 1 (their migration is underway, last I heard). They do use `get_credential` in their tests so it looks like this would break them -- could we include the track 2 check for the time being? I could understand an argument for dropping the test support at this point, but it's also not like there's any additional work to maintain the functionality for now.